akka-stream/src/main/scala/akka/stream/scaladsl/Source.scala

```diff
@@ -268,6 +268,20 @@ object Source {
     fromGraph(new FutureSource(future.toScala))
 
   /**
+   * A graph with the shape of a source logically is a source.
+   * This method makes an asynchronous graph of such shape in type with
+   * an asynchronous materialized value.
+   */
+  def fromFutureGraph[T, M](future: Future[Graph[SourceShape[T], M]]): Source[T, Future[M]] = fromGraph(new FutureFlattenSource(future))
+
+  /**
+   * A graph with the shape of a source logically is a source.
+   * This method makes an asynchronous graph of such shape in type with
+   * an asynchronous materialized value.
+   */
+  def fromGraphCompletionStage[T, M](future: CompletionStage[Graph[SourceShape[T], M]]): Source[T, CompletionStage[M]] = fromFutureGraph(future.toScala).mapMaterializedValue(_.toJava)
```

Comment:
`fromSourceCompletionStage` then, although no option for naming this that I can think of feels 100% good (mostly because `CompletionStage` is a crappy name to start with)